04 - Poker.playground/Contents.swift

```diff
@@ -7,16 +7,98 @@
 */
 /*
  EURE ANTWORT HIER
-*/
+ Wir verwenden hier Structs, da keine Vererbung wie bei Klassen benötigt wird. Structs sind gut um statische Dinge zu repräsentieren, wie z.B. Karten.
+ Enums verwenden wir, da die Einträge von Suit und Rank immer nur 4 bzw. 13 Möglichkeiten bieten, die sich gegenseitig ausschließen.
+ */
+enum Suit: Int, CustomStringConvertible{
+    case Diamonds, Spades, Hearts, Clubs
+    
+    var description: String{
+        switch self{
+            case .Diamonds:
+                return "diamond"
+            case .Spades:
+                return "spades"
+            case .Hearts:
+                return "hearts"
+            case .Clubs:
+                return "clubs"
+        }
+    }
+}
+    
+enum Rank: Int, CustomStringConvertible{
+    case two, three, four, five, six, seven, eight, nine, ten, jack, queen, king, ace
+    
+    var description: String{
+        switch self{
+        case .two:
+            return "two"
+        case .three:
+            return "three"
+        case .four:
+            return "four"
+        case .five:
+            return "five"
+        case .six:
+            return "six"
+        case .seven:
+            return "seven"
+        case .eight:
+            return "eight"
+        case .nine:
+            return "nine"
+        case .ten:
+            return "ten"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        case .ace:
+            return "ace"
+        }
+    }
+}
+/*extension Card: Equatable {}
+func ==(lhs: Card, rhs:Card) -> Bool {
+    return lhs.suit == rhs.suit && lhs.rank == rhs.rank
+ 
+}*/
 
+struct Card: CustomStringConvertible{
+    let suit: Suit
+    let rank: Rank
+    
+    
+    var description: String{
+        return "\(suit),\(rank)"
+    }
```

Comment:
Die `description` eines `CustomStringConvertible` Typs wird ausgegeben wenn du eine Instanz des Typs mit `print`est, also z.B.
```swift
let card = Card(suit: .Diamonds, rank: .two)
print(card)
```
Die Ausgabe erscheint dann in der Konsole.